interview-questions.md

```diff
@@ -8,63 +8,112 @@ Then, research the question to expand on your answer. Even if you feel you have
 
+  Researched answer:
+    A function can call itself recursively in the function body. A function is an object, and these functions as objects can have methods. A function within a function can access all of the variables within its parenet function and also any variable the parent function can access. You need a return statement right before the closing curly brackets.
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
+    Similarities:
+      .map() and .filter() are both higher order functions that serve a purpose within other functions. This means that they are like methods that can be added into the logic of your main function so that a certain secondary process can be automated to your values.
+    Differences:
+      .map() works as a for-loop would. It iterates through each value of an array, and you would add some functionality to this loop to change the given values. The length of the resulting array of .map() should be as long as the given array, as nothing is taken out of the array, only builded and iterated upon.
+      
+      .filter() also iterates through a given array, but as the name suggests, its purpose is to sort the array's values based on a given parameter. The resulting array should be shorter than the given array since .filter() removes values that do not fit the filter criteria.
+
+      The difference between .map() and .filter() are that .filter()'s resulting array will be shorter than the given array, while .map()'s resulting array should be the same length of the given array.
 
   Researched answer:
+    .map() wil return the same amount of elements as was given, but the new values will all have been transformed in some way. .filter() will return either the same or smaller amount of elemnts as was given, but the values will not have been changed.
 
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+    A function will run as long as it is evoked, but the result is not necessarily saved or shown to the user/coder unless it is programmed to do so. Return makes the resulting value of a function available to later pieces of code so that the value can be reused within its scope. A return does not inherently print itself to be shown to the user/coder. If that is the desired effect, the coder can console.log(myFunction(certainParameter)) so that the console can visibly print the results of the function with a certain parameter. Some consoles we have used in class include the terminal and the one found in Chrome. The difference between console.log and return is that the return is only visible to the machine, while a console.log makes the result available to the coder/user.
```

Comment:
Excellent answer, again, I think this would be JUST a bit long when answering in an interview setting.